lib/handlebars/compiler/compiler.js

```diff
@@ -323,18 +323,28 @@ Compiler.prototype = {
 
   Hash: function(hash) {
     let pairs = hash.pairs,
-        i = 0,
+        i = 0, splat = null,
         l = pairs.length;
 
     this.opcode('pushHash');
-
     for (; i < l; i++) {
+
+      if (pairs[i].type === 'Splat') {
+        if (splat !== null) {
+          throw new Exception('Multiple splats are not supported yet');
```

Comment:
@nathanhammond I was going to say that it seems more efficient to have the compiler do as much as possible to minimize the POJOs generated in favor of fewer POJOs with more props that we pass into merge, but I'm not actually sure if that's true (might be worth a JSPerf to see how that's optimized). 

Either way I'm nearly done wrapping up my first draft of the implementation, which leans toward minimizing POJOs.